byway/src/main/webapp/routepage.js

```diff
@@ -56,42 +70,75 @@ function initMap() {
  * @param {String} end ending point of route. TODO: In the final implementation, just have 
  * start as start and end are the same
  */
-function calcRoute(directionsService, directionsRenderer, start, end) {
+
+function calcRoute() {
   let request = {
     origin:  start,
     destination: end,
-    travelMode: 'DRIVING'
+    travelMode: 'DRIVING',
+    waypoints: Array.from(waypoints),
```

Comment:
Is waypoints supposed to be ordered? Sets generally don't guarantee any kind of ordering.

Reply:
Yeah, I made them sets so duplicate ones don't get added but I think I need ordering now that I want to use the optimize feature.